src/main/java/uk/gov/companieshouse/api/accounts/validation/ApprovalValidator.java

```diff
@@ -31,7 +49,29 @@ public Errors validateApproval(Approval approval, HttpServletRequest request) {
                 ErrorType.VALIDATION.getType()));
         }
 
+        ResponseObject<Director> directorsReportResponseObject = directorService.findAll(transaction, companyAccountId, request);
+
+        if(directorsReportResponseObject != null) {
+
+            Director[] directors = Optional.of(directorsReportResponseObject)
+                    .map(ResponseObject::getDataForMultipleResources)
+                    .orElse(null);
+
+            List<String> servingDirectors = new ArrayList<>();
+
+            if (directors != null) {
+
+                Arrays.stream(directors).filter(d -> d.getResignationDate() == null).forEach(director -> servingDirectors.add(director.getName()));
+            }
+
+            if (servingDirectors != null && !servingDirectors.isEmpty()) {
```

Comment:
You don't need the null check here; you instantiated `servingDirectors` so it can never be null

Reply:
Removed the extra null check that was not required